components/form.js

```diff
@@ -209,6 +230,28 @@ export function MarkdownInput ({ label, topLevel, groupClassName, onChange, setH
     }
   }, [innerRef, helpers?.setValue, setSelectionRange, onKeyDown])
 
+  const onDrop = useCallback((event) => {
+    event.preventDefault()
+    setDragStyle(null)
+    const changeEvent = new Event('change', { bubbles: true })
+    imageUploadRef.current.files = event.dataTransfer.files
+    imageUploadRef.current.dispatchEvent(changeEvent)
+  }, [imageUploadRef])
+
+  const [dragStyle, setDragStyle] = useState(null)
+  const onDragEnter = useCallback((e) => {
+    setDragStyle('over')
+  }, [setDragStyle])
+  const onDragLeave = useCallback((e) => {
+    setDragStyle(null)
+  }, [setDragStyle])
+
+  useEffect(debounce(() => {
+    const text = innerRef?.current.value
+    const s3Keys = text ? [...text.matchAll(AWS_S3_URL_REGEXP)].map(m => Number(m[1])) : []
+    updateImageFeesInfo({ variables: { s3Keys } })
+  }, 1000), [innerRef?.current?.value])
+
```

Comment:
Is there a way to do this without relying on a side effect of a reference value changing? 

Might it be simpler to call this in `onChange`?

Reply:
Ah yes, `debounce` isn't limited to use in `useEffect`, lol

Reply:
Done in 44d60f1